src/protocols/tcp6.rs

```diff
@@ -0,0 +1,178 @@
+//! Transmission Control Protocol version 6
+//!
+//! It provides services to send and receive data stream.
+
+use crate::efi::{Boolean, Ipv6Address};
+use crate::protocols::{ip6, managed_network, simple_network};
+
+pub const PROTOCOL_GUID: crate::base::Guid = crate::base::Guid::from_fields(
+    0x46e44855,
+    0xbd60,
+    0x4ab7,
+    0xab,
+    0x0d,
+    &[0xa6, 0x79, 0xb9, 0x44, 0x7d, 0x77],
+);
+
+#[repr(C)]
+pub struct AccessPoint {
+    pub station_address: Ipv6Address,
+    pub station_port: u16,
+    pub remote_address: Ipv6Address,
+    pub remote_port: u16,
+    pub active_flag: Boolean,
+}
+
+#[repr(C)]
+pub struct Tcp6Option {
+    pub receive_buffer_size: u32,
+    pub send_buffer_size: u32,
+    pub max_syn_back_log: u32,
+    pub connection_timeout: u32,
+    pub data_retries: u32,
+    pub fin_timeout: u32,
+    pub time_wait_timeout: u32,
+    pub keep_alive_probes: u32,
+    pub keep_alive_time: u32,
+    pub keep_alive_interval: u32,
+    pub enable_nagle: Boolean,
+    pub enable_time_stamp: Boolean,
+    pub enable_window_scaling: Boolean,
+    pub enable_selective_ack: Boolean,
+    pub enable_path_mtu_discovery: Boolean,
+}
+
+#[repr(C)]
+pub struct ConfigData {
+    pub traffic_class: u8,
+    pub hop_limit: u8,
+    pub access_point: AccessPoint,
+    pub control_option: *mut Tcp6Option,
+}
+
+pub type ConnectionState = u32;
+
+pub const CONNECTION_CLOSED: ConnectionState = 0;
```

Comment:
I dropped the `CONNECTION_*` prefix, as it is not used in the other values below. I added the `STATE_*` prefix though, as that follows how we encode C-enums.